datus/storage/scoped_filter.py

```
@@ -0,0 +1,190 @@
+        return eq("subject_node_id", unique_ids[0])
+    return or_(*[eq("subject_node_id", sid) for sid in unique_ids])
+
+
+def _subject_condition_for_parts(parts: List[str], subject_tree) -> Optional[Node]:
+    """Build a filter for a single subject token using two-pass resolution.
+
+    Pass 1: treat all *parts* as a full subject path (descendant=True).
+    Pass 2 (fallback): if pass 1 yields nothing and len(parts) > 1,
+            treat parts[:-1] as the path and parts[-1] as the item name.
+    """
+    # Pass 1 – full path
```

Ruff (0.15.2)

[warning] Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)